### PR TITLE
Transpose lowering to TOSA

### PR DESCRIPTION
**Pull request by Andrew-Cain**
Add lowering for Transpose to TOSA by creating new matchAndRewrite rule based off the existing permute matchAndRewrite rule.

### CODE REVIEW

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -2264,6 +2264,48 @@ LogicalResult ConvertAtenOp<AtenPermuteOp>::matchAndRewrite(
...
+template <>
+LogicalResult ConvertAtenOp<AtenTransposeIntOp>::matchAndRewrite(
+    AtenTransposeIntOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a ranked tensor type
+  auto selfType = adaptor.self().getType().dyn_cast<RankedTensorType>();
+  if (!selfType)
+    return op.emitError(
+        "Only ranked tensor types with static shapes are currently supported");
```
Comment (pashu123) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:2276]: Just casting to RankedTensorTypes doesn't check for static shapes - also check for `selfType.hasStaticShape()`.

**Comment by ramiro050** [comment]
There seems to be an overlap in PRs: https://github.com/llvm/torch-mlir/pull/1017 @sjarus

**Comment by sjarus** [comment]
Yes I just noticed. I'd prefer to keep mine if there isn't a problem, since I'm queuing a number of legalizations.

**Comment by Andrew-Cain** [comment]
It seems like #1017 is equivalent... 
I will close this PR in line with @sjarus' preference for #1017.

**Comment by sjarus** [comment]
Thanks and sorry for the duplication @Andrew-Cain . We're very happy to see others contributing to TorchToTosa, despite this start! Currently I have the following bits queued up besides aten.transpose.int: 

aten.slice_tensor
aten.broadcast_to 

If you're interested in anything else, please feel free to ping me and proceed.

**Comment by powderluv** [comment]
@Andrew-Cain appreciate the PR. Feel free to ping @sjarus on what you are working on in TorchToTosa on Discord.

**Pull request closed without merging**